engine/src/main/java/org/terasology/world/block/BlockRegion.java

```diff
@@ -93,392 +103,309 @@
+    public BlockRegion setMax(int x, int y, int z) {
+        return this.set(this.minX, this.minY, this.minZ, x, y, z);
     }
 
-    /**
-     * Sets the maximum coordinate of the second block for <code>this</code> {@link BlockRegion}
-     *
-     * @param max the second coordinate of the second block
-     * @return this
-     */
     public BlockRegion setMax(Vector3ic max) {
         return this.setMax(max.x(), max.y(), max.z());
     }
 
     /**
-     * sets the maximum block for this {@link BlockRegion}
+     * Translate the maximum corner of the region by adding given {@code (dx, dy, dz)}.
      *
-     * @param maxX the x coordinate of the first block
-     * @param maxY the y coordinate of the first block
-     * @param maxZ the z coordinate of the first block
-     * @return this
+     * @param dx the number of blocks to add to the maximum corner on the x axis
+     * @param dy the number of blocks to add to the maximum corner on the y axis
+     * @param dz the number of blocks to add to the maximum corner on the z axis
+     * @return {@code this} (after modification)
+     * @throws IllegalArgumentException if the resulting region would become {@link #isValid() invalid}.
      */
-    public BlockRegion setMax(int maxX, int maxY, int maxZ) {
-        Preconditions.checkArgument(maxX >= this.minX || this.minX == Integer.MAX_VALUE);
-        Preconditions.checkArgument(maxY >= this.minY || this.minY == Integer.MAX_VALUE);
-        Preconditions.checkArgument(maxZ >= this.minZ || this.minZ == Integer.MAX_VALUE);
-        this.maxX = maxX;
-        this.maxY = maxY;
-        this.maxZ = maxZ;
-        return this;
+    public BlockRegion addToMax(int dx, int dy, int dz) {
+        return this.setMax(this.maxX + dx, this.maxY + dy, this.maxZ + dz);
+    }
+
+    public BlockRegion addToMax(Vector3ic dmax) {
+        return this.addToMax(dmax.x(), dmax.y(), dmax.z());
     }
 
     // -- size -------------------------------------------------------------------------------------------------------//
 
     /**
-     * Set the size of the block region from minimum the minimum corner.
+     * Set the size of the block region from the minimum corner.
      *
      * @param x the x coordinate to set the size; must be > 0
      * @param y the y coordinate to set the size; must be > 0
      * @param z the z coordinate to set the size; must be > 0
-     * @return this after modification
+     * @return {@code this} (after modification)
+     * @throws IllegalArgumentException if the size is smaller than or equal to 0 in any dimension
      */
     public BlockRegion setSize(int x, int y, int z) {
-        Preconditions.checkArgument(x > 0);
-        Preconditions.checkArgument(y > 0);
-        Preconditions.checkArgument(z > 0);
-        this.maxX = this.minX + x - 1;
-        this.maxY = this.minY + y - 1;
-        this.maxZ = this.minZ + z - 1;
-        return this;
+        return this.setMax(this.minX + x, this.minY + y, this.minZ);
```

Comment:
```suggestion
        Preconditions.checkArgument(x > 0);
        Preconditions.checkArgument(y > 0);
        Preconditions.checkArgument(z > 0);
        return this.setMax(this.minX + x - 1, this.minY + y - 1, this.minZ + z - 1);
```

Reply:
That's right 👍 However, we don't need the preconditions here as we are calling `setMax` which will check conditions (so no need to duplicate the code).